janusgraph-all/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinPlugin.java

```diff
@@ -14,54 +14,143 @@
+
+        CLASS_IMPORTS.add(ConsistencyModifier.class);
+        CLASS_IMPORTS.add(DefaultSchemaMaker.class);
+        CLASS_IMPORTS.add(EdgeLabelMaker.class);
+        CLASS_IMPORTS.add(Index.class);
+        CLASS_IMPORTS.add(JanusGraphConfiguration.class);
+        CLASS_IMPORTS.add(JanusGraphIndex.class);
+        CLASS_IMPORTS.add(JanusGraphManagement.class);
+        CLASS_IMPORTS.add(JanusGraphSchemaElement.class);
+        CLASS_IMPORTS.add(JanusGraphSchemaType.class);
+        CLASS_IMPORTS.add(JobStatus.class);
+        CLASS_IMPORTS.add(Mapping.class);
+        CLASS_IMPORTS.add(Parameter.class);
+        CLASS_IMPORTS.add(PropertyKeyMaker.class);
+        CLASS_IMPORTS.add(RelationTypeIndex.class);
+        CLASS_IMPORTS.add(RelationTypeMaker.class);
+        CLASS_IMPORTS.add(SchemaAction.class);
+        CLASS_IMPORTS.add(SchemaInspector.class);
+        CLASS_IMPORTS.add(SchemaManager.class);
+        CLASS_IMPORTS.add(SchemaStatus.class);
+        CLASS_IMPORTS.add(VertexLabelMaker.class);
+
+        CLASS_IMPORTS.add(GraphOfTheGodsFactory.class);
+        CLASS_IMPORTS.add(JanusGraphIoRegistry.class);
+
+        CLASS_IMPORTS.add(Instant.class);
+        CLASS_IMPORTS.add(Clock.class);
+        CLASS_IMPORTS.add(DayOfWeek.class);
+        CLASS_IMPORTS.add(Duration.class);
+        CLASS_IMPORTS.add(LocalDate.class);
+        CLASS_IMPORTS.add(LocalTime.class);
+        CLASS_IMPORTS.add(LocalDateTime.class);
+        CLASS_IMPORTS.add(Month.class);
+        CLASS_IMPORTS.add(MonthDay.class);
+        CLASS_IMPORTS.add(OffsetDateTime.class);
+        CLASS_IMPORTS.add(OffsetTime.class);
+        CLASS_IMPORTS.add(Period.class);
+        CLASS_IMPORTS.add(Year.class);
+        CLASS_IMPORTS.add(YearMonth.class);
+        CLASS_IMPORTS.add(ZonedDateTime.class);
+        CLASS_IMPORTS.add(ZoneId.class);
+        CLASS_IMPORTS.add(ZoneOffset.class);
+
+        ///////////
+        // ENUMS //
+        ///////////
+
+        Collections.addAll(ENUM_IMPORTS, Geo.values());
+        Collections.addAll(ENUM_IMPORTS, Text.values());
+        Collections.addAll(ENUM_IMPORTS, Multiplicity.values());
+        Collections.addAll(ENUM_IMPORTS, Cardinality.values());
+        Collections.addAll(ENUM_IMPORTS, ChronoUnit.values());
     }
 
-    @Override
-    public void pluginTo(final PluginAcceptor pluginAcceptor) {
-        pluginAcceptor.addImports(IMPORTS);
+    private static final ImportCustomizer IMPORTS = DefaultImportCustomizer.build()
+        .addClassImports(CLASS_IMPORTS)
+        .addEnumImports(ENUM_IMPORTS).create();
+
+    private static final JanusGraphGremlinPlugin instance = new JanusGraphGremlinPlugin();
+
+    public JanusGraphGremlinPlugin() {
+        super(NAME, IMPORTS);
+    }
+
+    public static JanusGraphGremlinPlugin instance() {
+        return instance;
+    }
+
+    public JanusGraphGremlinPlugin(String moduleName, Customizer... customizers) {
+        super(moduleName, customizers);
```

Comment:
Need to add method imports.

`private final static Set<Method> METHOD_IMPORTS = new LinkedHashSet<>();`

```
Stream.of(Geo.class.getMethods())
            .filter(m -> Modifier.isStatic(m.getModifiers())).forEach(METHOD_IMPORTS::add);
Stream.of(Text.class.getMethods())
            .filter(m -> Modifier.isStatic(m.getModifiers())).forEach(METHOD_IMPORTS::add);
```

`.addMethodImports(METHOD_IMPORTS)`